controller/appcontroller.go

```diff
@@ -1879,6 +1885,14 @@ func (ctrl *ApplicationController) newApplicationInformerAndLister() (cache.Shar
 			cache.NamespaceIndex: func(obj interface{}) ([]string, error) {
 				app, ok := obj.(*appv1.Application)
 				if ok {
+					// If the application is not allowed to use the project,
+					// log an error.
+					if ctrl.isAppNamespaceAllowed(app) {
+						if _, err := ctrl.getAppProj(app); err != nil {
+							ctrl.setAppCondition(app, ctrl.projectErrorToCondition(err, app))
+						}
+					}
+
```

Comment:
A quick question, should we move the validate destination within AppNamespaceAllowed check as well? Wondering if we need to separately check for it if the Application is not allowed to use the project.